opsdroid/parsers/event_type.py

```diff
@@ -8,34 +8,49 @@
 _LOGGER = logging.getLogger(__name__)
 
 
+async def match_event(event, event_opts):
+    event_type = event_opts.get("type", None)
+    event_entitites = event.entities.get("value", {})
+
+    if event_type:
+        # The event type can be specified with a string
+        if isinstance(event_type, str):
+            # pylint: disable=invalid-name
+            et = Event.event_registry.get(event_type, None)
+            if et is None:
+                raise ValueError(
+                    "{event_type} is not a valid opsdroid"
+                    " event representation.".format(event_type=event_type)
+                )
+                event_type = et
+
+        # TODO: Add option to match all subclasses as well
+        # if isinstance(event, event_type):
+        # pylint: disable=unidiomatic-typecheck
+        if type(event) is event_type:
+            event_value = event_opts.get("value", None)
+            entity_value = event_entitites.get("value", None)
```

Comment:
This feels a bit rigid here, you're only implementing the `"value"` option for your specific use case.

It would probably be better to iterate over the items in `event_opts` (except `type`) and check if the event has that entitiy and if the values match.

Reply:
Changes done.